src/Std/Data/DTreeMap/Internal/Impl/Query.lean

```diff
@@ -0,0 +1,319 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Std.Data.DTreeMap.Internal.Impl.Def
+import Std.Data.OrderAxioms.LawfulEqOrd
+
+/-!
+# Low-level implementation of the size-bounded tree
+
+This file contains the basic definition implementing the functionality of the size-bounded trees.
+-/
+
+set_option autoImplicit false
+set_option linter.all true
+
+universe u v w
+
+variable {α : Type u} {β : α → Type v} {γ : α → Type w} {δ : Type w} {m : Type w → Type w}
+
+namespace Std.DTreeMap.Internal
+
+namespace Impl
+
+/-- The size information stored in the tree. -/
+@[inline]
+def size : Impl α β → Nat
+  | inner sz _ _ _ _ => sz
+  | leaf => 0
+
+@[Std.Internal.tree_tac] theorem size_leaf : (Impl.leaf : Impl α β).size = 0 := rfl
+@[Std.Internal.tree_tac] theorem size_inner {sz k v l r} : (Impl.inner sz k v l r : Impl α β).size = sz := rfl
+
+/-- Returns `true` if the given key is contained in the map. -/
+def contains [Ord α] (k : α) (t : Impl α β) : Bool :=
+  match t with
+  | .leaf => false
+  | .inner _ k' _ l r =>
+    match compare k k' with
+    | .lt => contains k l
+    | .gt => contains k r
+    | .eq => true
+
+instance [Ord α] : Membership α (Impl α β) where
+  mem t a := t.contains a
+
+instance [Ord α] {m : Impl α β} {a : α} : Decidable (a ∈ m) :=
+  show Decidable (m.contains a) from inferInstance
```

Comment:
Is this actually required?

Reply:
I want to add `mem` lemmas for `Impl`, too, hence I will indeed need this as soon as I write lemmas. We didn't have `mem` lemmas for `Raw0` of the hash map but I find it nice and consistent to handle `mem` just like every other operation and prove their lemmas for `Impl`.